src/components/structures/IndicatorScrollbar.js

```diff
@@ -53,10 +70,30 @@ export default class IndicatorScrollbar extends React.Component {
         } else {
             this._scrollElement.classList.remove("mx_IndicatorScrollbar_bottomOverflow");
         }
+        if (hasLeftOverflow) {
+            this._scrollElement.classList.add("mx_IndicatorScrollbar_leftOverflow");
+        } else {
+            this._scrollElement.classList.remove("mx_IndicatorScrollbar_leftOverflow");
+        }
+        if (hasRightOverflow) {
+            this._scrollElement.classList.add("mx_IndicatorScrollbar_rightOverflow");
+        } else {
+            this._scrollElement.classList.remove("mx_IndicatorScrollbar_rightOverflow");
+        }
 
         if (this._autoHideScrollbar) {
             this._autoHideScrollbar.checkOverflow();
         }
+
+        if (this.props.trackHorizontalOverflow) {
+            this.setState({
+                // Offset from absolute position of the container
+                leftIndicatorOffset: hasLeftOverflow ? `${this._scrollElement.scrollLeft}px` : 0,
+
+                // Negative because we're coming from the right
+                rightIndicatorOffset: hasRightOverflow ? `-${this._scrollElement.scrollLeft}px` : 0,
```

Comment:
Not sure why you'd take the negative `scrollLeft` here? If it's negative won't assigning `style.right` to this in `render` just move the gradient out of the parent container on the right side? I'd expect the offset to be something like `scrollWidth - (scrollLeft + clientWidth)` fwiw. Might be missing something.

Reply:
`right` is relative to the viewport, which means that if you use a positive number (or zero) when scrolled slightly to the right, this happens:
![image](https://user-images.githubusercontent.com/1190097/55176110-2281c000-5146-11e9-8c4d-5eb76742b4e6.png)

So by using a negative offset, you're pushing the gradient against the right edge of the viewport by the offset of the scroll.

Reply:
Oh right, my bad, I thought it was relative to the scrollable content.